### PR TITLE
Add documention example for handling Clerk webhooks to README.md

### PR DESCRIPTION
**Pull request by conradhunter**
Add working example for handling Clerks Webhooks in a C#/.NET app

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,6 +112,119 @@ async onRequestInit({ requestInit }) {
 }
 ```
 
+If you're accepting webhooks from Clerk, you will need to validate the incoming webhook signature. This can be done like this:
```
Comment (Hawxy) [inline README.md:115]: This should be under a new heading and should mention which Svix package needs to be installed.

File: README.md [hunk 2]
```diff
@@ -108,6 +112,119 @@ async onRequestInit({ requestInit }) {
...
+{
+    private readonly IConfiguration _configuration;
+
+    public ClerkController(IConfiguration configuration)
+    {
+        _configuration = configuration;
+    }
+
+    [HttpPost("webhook")]
+    public async Task<IActionResult> ClerkWebhook()
```
Comment (Hawxy) [inline README.md:138]: Inject `IConfiguration` as a function parameter and then delete everything above, reduces the amount of noise being shown.

File: README.md [hunk 3]
```diff
@@ -108,6 +112,119 @@ async onRequestInit({ requestInit }) {
...
+    {
+        // Retrieve the Clerk webhook secret from the configuration
+        var clerkWebhookSecret = "your-webhook-secret-here";
```
Comment (Hawxy) [inline README.md:141]: If `IConfiguration` is being injected, show it as being used here.

File: README.md [hunk 4]
```diff
@@ -108,6 +112,119 @@ async onRequestInit({ requestInit }) {
...
+
+        // Read the request body
+        var json = await new StreamReader(HttpContext.Request.Body).ReadToEndAsync();
+        Console.WriteLine($"Payload: {json}");
```
Comment (Hawxy) [inline README.md:145]: Console logging shouldn't be in an example.

File: README.md [hunk 5]
```diff
@@ -108,6 +112,119 @@ async onRequestInit({ requestInit }) {
...
+            { "svix-id", svixId },
+            { "svix-timestamp", svixTimestamp },
+            { "svix-signature", svixSignature }
+        };
+
+        Event webhookEvent;
+        try
+        {
+            wh.Verify(json, headers); // Verify doesn't return an event, just verifies the signature
+            webhookEvent = JsonConvert.DeserializeObject<Event>(json); // Deserialize the JSON into an Event object
```
Comment (Hawxy) [inline README.md:174]: Replace Newtonsoft usage with STJ. Newtonsoft shouldn't be used for modern examples.

**Review by Hawxy — CHANGES_REQUESTED** [review]
This is a nice addition but needs a few tweaks before merging.

File: README.md [hunk 6]
```diff
@@ -108,6 +112,119 @@ async onRequestInit({ requestInit }) {
...
+        public string ExternalId { get; set; } = string.Empty;
+        public string First_Name { get; set; } = string.Empty;
+        public string Last_Name { get; set; } = string.Empty;
+        public List<ClerkEmailAddress> Email_Addresses { get; set; } = new List<ClerkEmailAddress>();
+    }
+
+    public class ClerkEmailAddress
+    {
+        public string Email_Address { get; set; } = string.Empty;
+    }
```
Comment (Hawxy) [inline README.md:224]: Properties shouldn't be initialized to string by default, show them as nullable or `null!` forgiving based on Clerk's user model. Use `PascalCase` & `JsonPropertyName` for underscored properties.

File: README.md (hunk 5 above)
Comment (conradhunter) [reply README.md]: Good point

File: README.md (hunk 6 above)
Comment (conradhunter) [reply README.md]: Good points, thanks. Just started with C# this week so these pointers definitely help, appreciate it :)

File: README.md [hunk 7]
```diff
@@ -108,6 +112,108 @@ async onRequestInit({ requestInit }) {
...
+
+    public class ClerkEmailAddress
+    {
+        [JsonPropertyName("email_address")]
+        public string? EmailAddress { get; set; }
+    }
+```
+
 If the requests are coming from an SSR environment (ie NextJS), then you can either use the session JWT via the `__session` cookie and forward it on, or use a JWT template to create a unique JWT for your scenario.
```
Comment (Hawxy) [inline README.md:217]: This text should be with the above section, looks like it got accidentally bumped.

**Review by Hawxy — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**